pkg/storage/mongo/cataloger.go

```diff
@@ -20,22 +20,46 @@ func (s *ModuleStore) Catalog(ctx context.Context, token string, pageSize int) (
 
 	fields := bson.M{"module": 1, "version": 1}
 
-	c := s.s.DB(s.d).C(s.c)
+	compositeQ := bson.D{q, fields}
+
+	c := s.s.Database(s.d).Collection(s.c)
+
+	documentCount, erro := c.Count(compositeQ)
```

Comment:
I am not sure if this required or not. If you have a more than 0 count, then you end up making two calls to the MongoDB thus degrading performance.

Reply:
Yeah I wasn't sure myself too. Somehow I thought it wouldn't matter much if the fields are indexed, do an early exit. Will use the length of `modules` array only.